src/App.tsx

```diff
@@ -26,70 +27,125 @@ type ReorderOptions = {
   isReversed: boolean,
 };
 
-// Use this function in the render method to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  visibleGoods.sort((n1, n2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return n1.localeCompare(n2);
 
-  return visibleGoods;
+      case SortType.LENGTH:
+        return n1.length - n2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  return (!isReversed) ? visibleGoods : visibleGoods.reverse();
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  render() {
+    const { sortType, isReversed } = this.state;
+    const goods = getReorderedGoods(goodsFromServer, { sortType, isReversed });
+
+    const handleclick = (type: SortType) => {
+      this.setState({ sortType: type });
+      this.forceUpdate();
+    };
+
+    const reverseClick = () => {
```

Comment:
```suggestion
    const handleSort = (type: SortType) => {
      this.setState({ sortType: type });
      this.forceUpdate();
    };

    const handleReverse = () => {
```
I think it's better to specify what your methods handle